sqlglotrs/src/tokenizer.rs

```diff
@@ -0,0 +1,642 @@
+use crate::settings::TokenizerSettings;
+use crate::token::{Token, TokenType};
+use crate::trie::{Trie, TrieResult};
+use pyo3::prelude::*;
+
+#[derive(Debug)]
+#[pyclass]
+pub struct Tokenizer {
+    settings: TokenizerSettings,
+    keyword_trie: Trie,
+}
+
+#[pymethods]
+impl Tokenizer {
+    #[new]
+    #[pyo3(signature = (settings=None))]
+    pub fn new(settings: Option<TokenizerSettings>) -> Tokenizer {
+        let settings = settings.unwrap_or(TokenizerSettings::default());
```

Comment:
ah i see, can we just get rid of the default here and make settings mandatory, then we don't need the duplication?

Reply:
I wanted to keep it just for initial stages, so it's easier to reproduce and debug issues directly in rust without involving the python side.

Reply:
ok, as long as we delete it in the future

Reply:
that's the plan

Reply:
I figured we should be able to test just fine without these. Removed default settings